.github/workflows/CI.yaml

```diff
@@ -0,0 +1,92 @@
+name: CI workflow
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'docker/**'
+      - 'graphenex/**'
+  
+  push:
+    branches:
+      - main
+    paths:
+      - 'docker/**'
+      - 'graphenex/**'
+
+
+jobs:
+
+  linter:
+    name: "Linter check"
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+
+      - name: 'flake8 python linter'
+        uses: py-actions/flake8@v2
+
+
+  setup_linux:
+    name: "Linux setup"
+    runs-on: ubuntu-latest
+    needs: linter
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+      
+      - name: update
+        run: |
+          sudo apt-get update
+          sudo apt-get upgrade
+
+      - name: 'install dependencies'
+        run: |
+          sudo apt install python3-pip
+          python -m pip install --upgrade pip
+
+      - name: 'install graphenex'
+        run: python3 -m pip install graphenex
```

Comment:
Is this installing grapheneX from the checked out repository or from PyPI?

Reply:
From what I understand the command from instruction uses pip, so I would bet on PyPI.

Reply:
As instruction command uses pip I would bet it's using PyPI. Would you rather have it clone this repo and run ./graphenex/__main__.py or still use pip command?

Reply:
I'd say let's run it from the repo without pip.